pyscf/ci/cisd.py

```diff
@@ -913,9 +979,29 @@ def _finalize(self):
         return self
 
     def get_init_guess(self, eris=None, nroots=1, diag=None):
-        # MP2 initial guess
+        '''
+        MP2 energy and MP2 initial guess(es) for CISD coefficients.
+
+        Kwargs:
+            eris : ccsd._ChemistsERIs (inheriting) object (poss diff for df)
+                Contains the various (pq|rs) integrals needed.
+            nroots : integer
+                Number of CISD solutions to be found.
+            diag : numpy array (1D)
+                e.g. CISD Hamiltonian diagonal in Slater determinant
+                space with HF energy subtracted.
+
+        Returns:
+            Tuple of float and numpy array or
+            tuple of float and list of numpy arrays (if nroots > 1)
+            MP2 energy and initial guess(es) for CISD coefficients.
+
+        '''
         if eris is None: eris = self.ao2mo(self.mo_coeff)
         nocc = self.nocc
+        # [TODO] in `make_diagonal` function it says:
```

Comment:
In ccsd code, eris.mo_energy was used to compute initial MP2 amplitude. IMO, it should be okey to generate MP2 amplitudes for initial guess. This is consistent with the treatment in MP2 code.

In cisd.make_diagonal function, I think the accurate way to compute Slater determinant would be the Fock matrix constructed from the reference wavefunction. eris.mo_energy may not be appropriate to compute the value of Slater determinant if it was the mo_energy of underlying scf object. Seems in many modules it was set to fock.diagonal already.